gtk4/sample/examples/application4/exampleapp.rb

```diff
@@ -0,0 +1,147 @@
+      <?xml version="1.0" encoding="UTF-8"?>
+      <interface>
+        <template class="ExampleAppWindow" parent="GtkApplicationWindow">
+          <property name="title" translatable="yes">Example Application</property>
+          <property name="default-width">600</property>
+          <property name="default-height">400</property>
+          <child type="titlebar">
+            <object class="GtkHeaderBar" id="header">
+              <child type="title">
+                <object class="GtkStackSwitcher" id="tabs">
+                  <property name="stack">stack</property>
+                </object>
+              </child>
+              <child type="end">
+                <object class="GtkMenuButton" id="gears">
+                  <property name="direction">none</property>
+                </object>
+              </child>
+            </object>
+          </child>
+          <child>
+            <object class="GtkBox" id="content_box">
+              <property name="orientation">vertical</property>
+              <child>
+                <object class="GtkStack" id="stack"/>
+              </child>
+            </object>
+          </child>
+        </template>
+      </interface>
+      TEMPLATE
+      set_template(data: template)
+      bind_template_child("stack")
+      bind_template_child("gears")
+    end
+  end
+
+  def initialize(application)
+    menu_ui = <<~MENU
+    <?xml version="1.0" encoding="UTF-8"?>
+    <interface>
+      <menu id="menu">
+        <section>
+          <item>
+            <attribute name="label" translatable="yes">_Preferences</attribute>
+            <attribute name="action">app.preferences</attribute>
+          </item>
+        </section>
+        <section>
+          <item>
+            <attribute name="label" translatable="yes">_Quit</attribute>
+            <attribute name="action">app.quit</attribute>
+          </item>
+        </section>
+      </menu>
+    </interface>
+    MENU
+    super(application: application)
+    builder = Gtk::Builder.new(string: menu_ui)
+    gears.menu_model = builder["menu"]
+  end
+
+  def open(file)
+    basename = file.basename
+    scrolled = Gtk::ScrolledWindow.new
+    scrolled.hexpand = true
+    scrolled.vexpand = true
+    view = Gtk::TextView.new
+    view.editable = false
+    view.cursor_visible = false
+    scrolled.child = view
+    stack.add_titled(scrolled, basename, basename)
+    stream = file.read
+    view.buffer.text = stream.read
+  end
+end
+
+class ExampleApp < Gtk::Application
+  def initialize
+    super("org.gtk.exampleapp", :handles_open)
+
+    signal_connect "startup" do |application|
+      [ # action_name, handler (handler can be a Method object or Proc object)
+        ["preferences"],
+        ["quit", application.method(:quit)]
+      ].each do |action_name, handler|
+        action = Gio::SimpleAction.new(action_name)
+        action.signal_connect("activate") do |_action, _parameter|
+          handler.call if handler
+        end
```

Comment:
How about simplify this?

```suggestion
      [
        "preferences",
        "quit",
      ].each do |action_name,|
        action = Gio::SimpleAction.new(action_name)
        action.signal_connect("activate") do |_action, _parameter|
          __send__("#{action_name}_activated")
        end
```

And define `_activated` methods:

```ruby
private
def preferences_activated
end

def quit_activated
  quit
end
```